### ISSUE TITLE
2D View Enhancements / Labeling

### ISSUE THREAD

**Issue by nailsonlandim** [CCDevNelson/Snap-Updates#164]
TODO

- [x] # 1 - For variable section/opening widths it should add a V to the end of the dimension (e.g., 20.25"V)

- [x] # 2 - Label countertop type on the drawing together with width and depth dimensions (melamine, HPL, quartz, etc., see examples)

- [x] # 3 - Add option to label "Build Height"--Build height is the hang/panel height plus the topshelf or countertop thickness.

- [x] # 4 - Move location of labels for partition height and build height to the side of the elevation (see examples)

- [x] # 5 - Change labeling of doors, drawers and hampers from a decimal number to hole number (3H, 4H, etc.) (see examples)

- [x] # 6 - Have labeling Toe Kick height be a main label rather than an add-on and have it be on the side of the elevation

- [x] # 7 - Label File Drawers (will also have to label jewelry drawers when that option is added)

- [T] # 8 - Fix angled lines--sometimes partition / wall lines aren't perfectly straight up and down but rather come out as a perspective/angled

- [x] # 9 - Wall height and wall width dimensions should automatically be put on both plan and elevation views (wall/ceiling height to be on the side of the elevation and the wall width above the elevation, on plan view the wall height should be notated at the corners of the room and at the corner of any soffit)

- [x] # 10 - Get rid of the rounding to the nearest fraction--should be rounded to the nearest hundredth decimal place

- [x] # 11 - Add full back (F/B) label in middle of every full back. Best to say 1/4" F/B" or 3/4" F/B if possible.

![image](https://user-images.githubusercontent.com/61025894/89741717-772a9000-da48-11ea-8398-88fd690c24fe.png)

![image](https://user-images.githubusercontent.com/61025894/89741751-be188580-da48-11ea-89d4-1aad96025dda.png)

EDIT: Item 8 became issue #227 

See also for reference:

https://github.com/CCDevNelson/SNaP-Dev/issues/222
https://github.com/CCDevNelson/SNaP-Dev/issues/108
https://github.com/CCDevNelson/SNaP-Dev/issues/40
https://github.com/CCDevNelson/SNaP-Dev/issues/38
https://github.com/CCDevNelson/SNaP-Dev/issues/77

**Comment by TEDSTANDAGE** [comment]
- [ ] For variable section/opening widths it should add a V to the end of the dimension (e.g., 20.25"V)

**Comment by nailsonlandim** [comment]
It seems I can group them as below (correct me if am I wrong):

- Add a label (optional or not)
**1, 2, 3, 7, 11** 

- Move or Change existing label/dimension
**4, 5, 6, 9**

- Fix 2D render
**8**

- Fix Dimension rounding/precision
**10**

Is this the problem on **8**?
![angled_lines](https://user-images.githubusercontent.com/70038700/94075205-280aa580-fdd1-11ea-80ac-a7e1b708290a.png)

**Comment by TEDSTANDAGE** [comment]
> It seems I can group them as below (correct me if am I wrong):
> 
> * Add a label (optional or not)
>   **1, 2, 3, 7, 11**
> * Move or Change existing label/dimension
>   **4, 5, 6, 9**
> * Fix 2D render
>   **8**
> * Fix Dimension rounding/precision
>   **10**

Yes that's fine to group them that way

> 
> Is this the problem on **8**?
> ![angled_lines](https://user-images.githubusercontent.com/70038700/94075205-280aa580-fdd1-11ea-80ac-a7e1b708290a.png)

Yes that is the issue.

Thank you

**Comment by TEDSTANDAGE** [comment]
> Is this the problem on **8**?
> ![angled_lines](https://user-images.githubusercontent.com/70038700/94075205-280aa580-fdd1-11ea-80ac-a7e1b708290a.png)

Sometimes vertical wall lines are missing entirely from the 2D render as well. I will try to find an example.

**Comment by nailsonlandim** [comment]
After a few tests, I could notice that on custom rooms my recent changes caused duplication of dimensions.

Add dimensions on a custom room by default completely is needed (as it is custom), so I ask @RyanMDev if there is a way to check if a dimension already exists, to avoid this:

![Captura de tela 2020-09-29 163906](https://user-images.githubusercontent.com/70038700/94607477-588b8d00-0272-11eb-8e50-a915a4fcce4b.png)

**Comment by TEDSTANDAGE** [comment]
Good start, thank you Nailson. Like we discussed via Skype, let's see if we can change the orientation of the dimensions of the partitions to parallel the partitions instead of being perpendicular. Also as we discussed, changing it to text instead of dimensions is fine--so long as the text updates if someone makes a change to the closet. Accurate info just needs to end up on the 2D View, the way it gets there isn't that important to me as long as it's ultimately accurate.

**Comment by TEDSTANDAGE** [comment]
However, changing the orientation is not priority--let's get the rest of these things fixed and we can work on the even more finer details later. Thanks, Ted

**Comment by nailsonlandim** [comment]
Hi @TEDSTANDAGE , How do you want the wall heights on plan view? We have some freedom.

![164_item_9-1](https://user-images.githubusercontent.com/70038700/94879965-1a3fca80-0438-11eb-8f71-1627e21b1163.png)

**I also added them to elevations**

![164_item_9-3](https://user-images.githubusercontent.com/70038700/94880120-81f61580-0438-11eb-8562-48d803816a75.png)

You can check it at https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

**Comment by TEDSTANDAGE** [comment]
Thanks Nailson, please mimic the example:

![3A0A2B39-5984-4157-8BE7-4C08321E9BA9](https://user-images.githubusercontent.com/61025894/94926588-e08db480-0475-11eb-910f-6b97ab904f08.jpeg)

A hash from the corner out to the wall height dimension.

**Comment by nailsonlandim** [comment]
Hi @TEDSTANDAGE 

![164 Item 6](https://user-images.githubusercontent.com/70038700/94947894-c6bf9200-04b4-11eb-8aa6-6599dd866f03.png)

This is what you want? If so, I pushed it: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

**Comment by TEDSTANDAGE** [comment]
Yes, thank you. I might have you adjust the position later but that's good for now.

**Comment by nailsonlandim** [comment]
@TEDSTANDAGE I've added the file drawers label,

![File Drawers 1](https://user-images.githubusercontent.com/70038700/95484709-30351a00-0967-11eb-8ba8-4aa088a83bdf.png)

![File Drawers 2](https://user-images.githubusercontent.com/70038700/95484719-31fedd80-0967-11eb-9058-c1422259d99e.png)


The countertop labeling isn't working in every project. I'll check on that, Tell me if it's labeling position is good as expected. I appreciate any comment.

**Comment by TEDSTANDAGE** [comment]
Thank you Nailson, they look off center like they need to be moved to the left a bit. Can you fix that? Thanks, Ted

**Comment by nailsonlandim** [comment]
> Thank you Nailson, they look off center like they need to be moved to the left a bit. Can you fix that? Thanks, Ted
> […](#)
> Sent from my iPhone
> On Oct 8, 2020, at 9:13 AM, Nailson Landim ***@***.***> wrote: ﻿ @TEDSTANDAGE I've added the file drawers label, The countertop labeling isn't working in every project. I'll check on that, Tell me if it's labeling position is good as expected. I appreciate any comment. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Sure, these adjustments are simple, I`ll take care of it.

**Comment by nailsonlandim** [comment]
Finished Item 2, Also adjusted the file drawer label a bit to left, On Blender, 2D View stays almost perfect, at the PDF it goes to the right.  

![](https://user-images.githubusercontent.com/70038700/95515221-a3ed1c00-0993-11eb-87e8-0f87686458e9.png)

![image](https://user-images.githubusercontent.com/70038700/95515914-c16eb580-0994-11eb-9818-6e7b8741d5d2.png)

**Comment by nailsonlandim** [comment]
**I could also add the Fullback label, just dealing with it's positioning**

![image](https://user-images.githubusercontent.com/70038700/95522678-19acb400-09a3-11eb-81d7-38c9a75e056f.png)

**Comment by nailsonlandim** [comment]
@TEDSTANDAGE Finished Item 1, This time I could center it better. Tell me if you want any adjustments.

![Variable Sec 01](https://user-images.githubusercontent.com/70038700/95640405-05d08300-0a73-11eb-8f0e-32e044ae5c09.png)

![Variable Sec 02](https://user-images.githubusercontent.com/70038700/95640410-09640a00-0a73-11eb-8db2-1b249d832dec.png)

It's available at the same branch: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

**Comment by nailsonlandim** [comment]
Hi @TEDSTANDAGE, I did some adjustments as you requested. They stay where you want and have the Var. tag


![image](https://user-images.githubusercontent.com/70038700/95866523-647d5180-0d3e-11eb-8935-3f4c2ecd4b21.png)

**Comment by nailsonlandim** [comment]
@TEDSTANDAGE Finished Item 3, Build Height. Tell me what you think.

There are a few dimensions that are overlaying, Tell me the order you want to show them. 

![Item 3 - Build Height - 01](https://user-images.githubusercontent.com/70038700/95927752-0b410c80-0d96-11eb-9ed5-00aeb6c3b5bf.png)

![Item 3 - Build Height - 02](https://user-images.githubusercontent.com/70038700/95927755-0c723980-0d96-11eb-95f6-ef0c03737525.png)

Correct me if I am wrong, but dimensions should be something similar to this:

![dimensions](https://user-images.githubusercontent.com/70038700/95928217-2eb88700-0d97-11eb-8683-391bf8194fe7.png)

**Comment by TEDSTANDAGE** [comment]
Thanks Nailson, 

In most situations a user would only use one out of the build height or hang height, not both, so they wouldn't overlap each other like your example. If they did choose both then the heights would likely be different enough that the numbers themselves at least wouldn't overlap. Ideally, the dimension line would have a break in it like the following example which also would help avoid the overlap:

![image](https://user-images.githubusercontent.com/61025894/95997281-7c111480-0de8-11eb-9b09-9a89e724fd95.png)

As far as order, it would go from the tallest to the shortest with the tallest being on the outside and the shortest on the inside.

**Comment by TEDSTANDAGE** [comment]
> **I could also add the Fullback label, just dealing with it's positioning**
> 
> ![image](https://user-images.githubusercontent.com/70038700/95522678-19acb400-09a3-11eb-81d7-38c9a75e056f.png)
> 
> EDIT:
> 
> I still haven't figured a way to finely position an F/B label on the Z-axis.
> 
> X and Y-axis isn't a problem. but even if I add an arbitrary value on Z, it doesn't work (like +30 inches), they stay at the same position. Any thoughts @RyanMDev @ClassyTeddy ?

@ClassyTeddy @RyanMDev any help or ideas for Nailson on positioning the f/b label? Ideally it should be centered right in the middle of the f/b on the 2D views. Thanks if you can.

**Comment by nailsonlandim** [comment]
In fact, 2D Views works with X or Y and Z axis. It depends on which wall is rendering. It is counterintuitive, but it how it works

![image](https://user-images.githubusercontent.com/70038700/96000721-97dae180-0e0d-11eb-905c-585942d61ab8.png)

![image](https://user-images.githubusercontent.com/70038700/96000875-c0fb7200-0e0d-11eb-9e75-079c83a7952c.png)

In fact, I got an idea @TEDSTANDAGE @ClassyTeddy . 

As it's not necessary to display the dimension, but it's a label I'll insert it as a Blender Text object.

**Comment by TEDSTANDAGE** [comment]
> 
> In fact, I got an idea @TEDSTANDAGE @ClassyTeddy .
> 
> As it's not necessary to display the dimension, but it's a label I'll insert it as a Blender Text object.

I think that's a good idea--Would that make it easier for the user to move the label around after the label is placed into the scene in case it overlaps/conflicts with something else there in the opening? I have been wondering about ways to make it easier for the user to move around dimensions/labels after the label is placed but before the 2D view is prepared as it can get messy quick (here is some of my messing around):

![image](https://user-images.githubusercontent.com/61025894/96001679-255a0980-0ded-11eb-84cc-dee4aea78bba.png)

**Comment by nailsonlandim** [comment]
> I think that's a good idea--Would that make it easier for the user to move the label around after the label is placed into the scene in case it overlaps/conflicts with something else there in the opening? I have been wondering about ways to make it easier for the user to move around dimensions/labels after the label is placed but before the 2D view is prepared as it can get messy quick (here is some of my messing around):

Yes, you can move around Text objects easily. That will help in those situations.

![image](https://user-images.githubusercontent.com/70038700/96002438-7bd83f80-0e0f-11eb-84bc-dfc9c8b6c3f1.png)

After your reply earlier about the dimensions I think that I will "collect" the dimensions and displace them in order, so it will help it become a mess. So if I have three dimensions to display it will arrange them.

**Comment by nailsonlandim** [comment]
@TEDSTANDAGE Finished Item 11, 

![image](https://user-images.githubusercontent.com/70038700/96033703-d1741280-0e36-11eb-8050-294c9d8f727a.png)

**You can move them...**

![image](https://user-images.githubusercontent.com/70038700/96033921-24e66080-0e37-11eb-97e3-583c8d3748ba.png)

**... and get the result you want to:**

![image](https://user-images.githubusercontent.com/70038700/96033846-054f3800-0e37-11eb-8ce8-f67463f99ddc.png)

Available at: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

**Comment by TEDSTANDAGE** [comment]
﻿Great, thank you Nailson. Looks good. Is it possible to match the f/b text to the other dimensions/labels? It seems a little out of place.

Sent from my iPhone

**Comment by nailsonlandim** [comment]
> ﻿Great, thank you Nailson. Looks good. Is it possible to match the f/b text to the other dimensions/labels? It seems a little out of place.

I place them right in the middle of the FB.

As they are Texts, the Designer can move them if wanted. If you want it otherwise, just tell me.

**Comment by TEDSTANDAGE** [comment]
Sorry, by “out of place” I didn’t mean location. I meant the text style/size—it doesn’t match the rest of the dimensions/labels. The FB label is bolder and a little bigger than the other dimensions/labels

Sent from my iPhone

**Comment by nailsonlandim** [comment]
> Sorry, by “out of place” I didn’t mean location. I meant the text style/size—it doesn’t match the rest of the dimensions/labels. The FB label is bolder and a little bigger than the other dimensions/labels

Understood, I will change them to look as similar as dimensions.

**Comment by nailsonlandim** [comment]
@TEDSTANDAGE I checked and they are the same font (Calibri-light), but Blender Text renders in a different way than the OpenGL Dimension. I changed it a bit, but without major changes. 

I also would enjoy it if it could look like other texts.

![image](https://user-images.githubusercontent.com/70038700/96138234-ef438500-0ed3-11eb-8739-956854feea77.png)

**Comment by nailsonlandim** [comment]
@TEDSTANDAGE Finished Item 10, rounding to the nearest hundredth decimal place

![image](https://user-images.githubusercontent.com/70038700/96181045-bf16d900-0f09-11eb-9197-314c9ec37e9e.png)

I also used the opportunity to change the UI, I switched the "Round to the nearest" select box for the Inches option to a Precision box. You can choose it if you want to, using 2 digits by default

**Comment by nailsonlandim** [comment]
Hi @TEDSTANDAGE , I've changed Drawers and Hampers to hole number. 

![image](https://user-images.githubusercontent.com/70038700/96275352-018ef300-0fa8-11eb-9b5a-13eb30676067.png)

Tell me if you want it in a different way for hampers or if it works.

**Comment by TEDSTANDAGE** [comment]
Thank you Nailson. Please have the drawer hole label vertically be centered on the drawer face and move it to the right side of the face (centered between the handle and the right edge of the face), then have the hamper hole label be on the upper right hand of the hamper face. Thanks, Ted

**Comment by nailsonlandim** [comment]
Hi @TEDSTANDAGE , Finished it, Doors are labeled in hole number, accordingly the 0.25" tolerance. 

![image](https://user-images.githubusercontent.com/70038700/96321252-13de5080-0feb-11eb-9487-66a0a48cdcb4.png)

Tell me if you have any thought or idea.

**Comment by TEDSTANDAGE** [comment]
Looking really good Nailson. Here are a few notes of some tweaks to make:

![Label review](https://user-images.githubusercontent.com/61025894/96504498-9a707900-1209-11eb-9876-9c016a49159a.png)

Thanks, Ted

**Comment by TEDSTANDAGE** [comment]
Also, in the above example, it incorrectly states that the build height is 83.0 when it should be 84.0--the panel hang height is correctly shown as 83.25", then with the 0.75" thick topshelf the build height is 84.0. I'm not sure what's going on with that.

**Comment by TEDSTANDAGE** [comment]
Also, let's have 1/4" F/B and 3/4" F/B be written on two lines instead of one:

1/4"
F/B

Not 1/4" F/B

3/4"
F/B

Not 3/4" F/B

**Comment by nailsonlandim** [comment]
Hi @TEDSTANDAGE , I've worked on the corrections you've asked for.

- Removed the hampers labels appearing on Plan View
- Adjusted the Hamper face label
- Adjusted the fullback text
- Tuned the partition dimension to stay within the opening and placed the text parallel to it
- Fixed the Build Height dimension
- Started the arrangement on Build Height / Wall Height. 

The heights on left are colliding with each other. I can add more space or place them in parallel as I did with partitions. 

![image](https://user-images.githubusercontent.com/70038700/96652561-8e6ddf80-130d-11eb-8f17-b7d023cca6e1.png)

Tell me how you want the Countertop label...
![image](https://user-images.githubusercontent.com/70038700/96654614-19e96f80-1312-11eb-9acb-b24c505a8ce1.png)

I'm working on the double door fix yet, It will be easy.

**Comment by TEDSTANDAGE** [comment]
Thanks Nailson. For the hamper face, I didn't want the hamper type label moved (e.g., the word wire). I wanted the hole number moved to the left (centered between the handle and the  to line up with the drawer face hole number label:

![line up](https://user-images.githubusercontent.com/61025894/96725347-a6714d80-1365-11eb-83ff-23528bce0374.png)

Then, the drawer face labels do not look to be centered vertically on the drawer face (see how the label 3H is overlapping the top line of the drawer face? It should fit within the drawer face). The drawer face handles are already centered vertically on the drawer face (except on 10H+ drawers) so you should just match that. You can make the same exception for 10H+ drawer faces as well and move the label up accordingly.

Also, can you have the partition dimensions be on the left side of the partition like it was before rather than the right side? I know you were trying to be within the opening but that's not how we normally do it.

Yes, please change the orientation of the build height and the wall height so that they do not overlap each other. You can get rid of the hang height option--the build height will just be the hang height if there is no top shelf or countertop on top of the partition, correct? Then you won't have those two dimensions overlapping eachother like they are right now. Let me know if you have any questions.

Thanks, Ted

**Comment by nailsonlandim** [comment]
Hi @TEDSTANDAGE 

I aligned the drawer/hampers/doors labels as you requested. 

I vertically centered the drawers by adding an offset for 3 to 9H and another for 10H. 
A more refined possibility, if you need to, is by looking for the drawer pull height (if exists). It's a possibility but IMHO it's not necessary. 

I also worked on the double door labeling, I search for a sibling door. 

![image](https://user-images.githubusercontent.com/70038700/96754607-e9541500-13a7-11eb-87ee-8c1a6b77a4d1.png)

I am getting rid of the hang height, and yes you're right, the build height will be the hang height if there's no countertop or topshelf.

![image](https://user-images.githubusercontent.com/70038700/96754685-012b9900-13a8-11eb-8e45-a09015eac428.png)

I still a bit confused about the partitions dimensions, I changed to the previous state (expect for the label), I can set them a bit more to left if you want to. 

Available at: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

**Comment by TEDSTANDAGE** [comment]
Thank you Nailson, the location of the hole size labels look great. As far as moving the dimensions to the left side of the partition rather than the right (and also please do this for the build height and wall height that you have moved to be parallel), please turn them 180 degrees and move to the other side of the partition/dimension than what you currently have. Here is what they should look like:

![flip dimensions](https://user-images.githubusercontent.com/61025894/96760737-d6344b80-138d-11eb-9235-385df5fe4fac.png)

Let me know if you have any other questions.

**Comment by nailsonlandim** [comment]
Hi, @TEDSTANDAGE 

I made the changes and got rid of hang height.

![image](https://user-images.githubusercontent.com/70038700/96778748-799f5380-13c2-11eb-89fd-0d48eb3a267e.png)

Thanks, 
Nailson

**Comment by TEDSTANDAGE** [comment]
@sarahhightower @kategodfrey @JBTattersall Can you all look at all the changes Nailson has made (remember it is in his repository) and then let's get together to discuss the tweaks and additions we'd like to make? He has made a lot of improvements and I know every designer has their own way of doing things so we all might not agree with the way I have had him lay everything out. Let's discuss between us before comments/direction is made to Nailson so we can give him clear direction from one voice. Thanks, Ted

**Comment by nailsonlandim** [comment]
Hi @sarahhightower @kategodfrey @JBTattersall , I've added the hash marks @TedClassy1209 asked me to do.

I changed the countertop labeling into two lines to fit it better. In some scenarios, it had the same length as the countertop. So now it fits the label and the line without problems.

![image](https://user-images.githubusercontent.com/70038700/97504861-88ec4700-1956-11eb-845b-76bcba5eb598.png)

Plan views also received their marks...
![image](https://user-images.githubusercontent.com/70038700/97504902-9f929e00-1956-11eb-8468-b05d12449227.png)

It's available at: 
https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

**Comment by TedClassy1209** [comment]
Thanks, great work. A few things—can you have the hash mark touch the corner or shelf or the exposed end it is referring to? Looks like it is hanging out in space in a few instances. 

Also the hash mark should default to go up and to the right, so please switch the Topshelf hash mark. 

Please change the word exposed to “Exp.”

Can you change the plan view to only have one wall/ceiling height notated instead of every single room (unless there are different ceiling heights within the closet) The top right corner of the plan view is preferred, though I know that’ll change which wall needs it if it is a custom room.

**Comment by nailsonlandim** [comment]
> Thanks, great work. A few things—can you have the hash mark touch the corner or shelf or the exposed end it is referring to? Looks like it is hanging out in space in a few instances.

Sure, I can work on that.

> Also the hash mark should default to go up and to the right, so please switch the Topshelf hash mark.

Understood

> Please change the word exposed to “Exp.”

Ok

> Can you change the plan view to only have one wall/ceiling height notated instead of every single room (unless there are different ceiling heights within the closet) The top right corner of the plan view is preferred, though I know that’ll change which wall needs it if it is a custom room.

Ok, I'll think of a solution for custom rooms

**Comment by TedClassy1209** [comment]
Also to save space you can just use "C.T." instead of spelling out countertop and "Mel." instead of spelling out Melamine.

**Comment by TedClassy1209** [comment]
For the section depths, please follow the examples and have it coming off the bottom of the opening with the hash mark going up and to the right to the dimension.

**Comment by nailsonlandim** [comment]
Hi @TedClassy1209 

I did the adjustments you asked for. I am working on Plan view wall heights.

![image](https://user-images.githubusercontent.com/70038700/97713933-43876100-1a9f-11eb-8ea8-89e5121fa36f.png)

**Comment by sarahhightower** [comment]
Hi Guys! 

I just got the Nailson repository and tested this out.  I love all the progress that's been made! I'm including my 2D views and my feedback for continued development below. 

* The EXP appears in places where it's not needed, and missing from the places that it is.  On Wall A it should only appear on the right end of the shelving unit.  It does not need to appear anywhere on Wall B. And on Wall C it should also appear on left side of the enclosed 24"depth door cabinet but not on the right side of the double hang. 

* The shelving section on Wall A is missing the depth, which is different from the rest of Wall A at 16". 

* There are multiple places where I double pard-ed and this is not clear in the design. It would be great to have a vertical line below the double partition and then a "DP" to notate this. This is how it's typically noted on drawn plans. (The double pard-ed areas in my design are the 20"D drawer stack area on Wall B and the area to the left and right of the enclosed cabinet on Wall C.) 

* On Wall C all of my cabinets are 30" wide, but the 2D shows them as 29" wide. I'm not sure why? 

* I chose a granite countertop, but it still says that it's melamine.

* I have a "hang double tall" behind the enclosed cabinet on Wall C but that's not clear with the given 2D view.

* I have uppers and lowers with different widths on wall B and they appear on top of each other so you can't see either dimension clearly. Could the upper widths be placed right beneath the upper cabinets?

[2D Views.pdf](https://github.com/CCDevNelson/Snap-Updates/files/5467959/2D.Views.pdf)

Thanks!

Sarah

**Comment by TedClassy1209** [comment]
> Hi Guys!
> 
> I just got the Nailson repository and tested this out. I love all the progress that's been made! I'm including my 2D views and my feedback for continued development below.
> 
> * The EXP appears in places where it's not needed, and missing from the places that it is.  On Wall A it should only appear on the right end of the shelving unit.  It does not need to appear anywhere on Wall B. And on Wall C it should also appear on left side of the enclosed 24"depth door cabinet but not on the right side of the double hang.
> * The shelving section on Wall A is missing the depth, which is different from the rest of Wall A at 16".
> * There are multiple places where I double pard-ed and this is not clear in the design. It would be great to have a vertical line below the double partition and then a "DP" to notate this. This is how it's typically noted on drawn plans. (The double pard-ed areas in my design are the 20"D drawer stack area on Wall B and the area to the left and right of the enclosed cabinet on Wall C.)
> * On Wall C all of my cabinets are 30" wide, but the 2D shows them as 29" wide. I'm not sure why?
> * I chose a granite countertop, but it still says that it's melamine.
> * I have a "hang double tall" behind the enclosed cabinet on Wall C but that's not clear with the given 2D view.
> * I have uppers and lowers with different widths on wall B and they appear on top of each other so you can't see either dimension clearly. Could the upper widths be placed right beneath the upper cabinets?
> 
> [2D Views.pdf](https://github.com/CCDevNelson/Snap-Updates/files/5467959/2D.Views.pdf)
> 
> Thanks!
> 
> Sarah

Thank you Sarah, can you you also share the project file for that job? You can save it to Google Drive (the ccp file, blend file, etc.) then share the google drive link.

Nailson, please message me tomorrow when you are available and ready to talk about this--some of the items identified I don't think are with what you are working on (e.g., the granite countertop--the countertops are all showing in the scene as melamine now even if quartz or granite is chosen, but I will have Teddy look into that issue) and some of the solutions to some of the other items like the DP issue will be taken care of by other existing outstanding issues. Also Sarah Katie Jim and I are hopefully meeting later this week to discuss and provide further direction for you.

Thanks, Ted

**Comment by sarahhightower** [comment]
Oh sure I can do that.  Sorry if I gave feedback that didn’t make sense for the situation. ☹ Sarah HightowerClassy Closets602-373-6103 From: TheoSent: Monday, November 2, 2020 9:35 AMTo: CCDevNelson/Snap-UpdatesCc: sarahhightower; MentionSubject: Re: [CCDevNelson/Snap-Updates] 2D View Enhancements / Labeling (#164) Hi Guys!I just got the Nailson repository and tested this out. I love all the progress that's been made! I'm including my 2D views and my feedback for continued development below.The EXP appears in places where it's not needed, and missing from the places that it is. On Wall A it should only appear on the right end of the shelving unit. It does not need to appear anywhere on Wall B. And on Wall C it should also appear on left side of the enclosed 24"depth door cabinet but not on the right side of the double hang.The shelving section on Wall A is missing the depth, which is different from the rest of Wall A at 16".There are multiple places where I double pard-ed and this is not clear in the design. It would be great to have a vertical line below the double partition and then a "DP" to notate this. This is how it's typically noted on drawn plans. (The double pard-ed areas in my design are the 20"D drawer stack area on Wall B and the area to the left and right of the enclosed cabinet on Wall C.)On Wall C all of my cabinets are 30" wide, but the 2D shows them as 29" wide. I'm not sure why?I chose a granite countertop, but it still says that it's melamine.I have a "hang double tall" behind the enclosed cabinet on Wall C but that's not clear with the given 2D view.I have uppers and lowers with different widths on wall B and they appear on top of each other so you can't see either dimension clearly. Could the upper widths be placed right beneath the upper cabinets?2D Views.pdfThanks!SarahThank you Sarah, can you you also share the project file for that job? You can save it to Google Drive (the ccp file, blend file, etc.) then share the google drive link.Nailson, please message me tomorrow when you are available and ready to talk about this--some of the items identified I don't think are with what you are working on (e.g., the granite countertop--the countertops are all showing in the scene as melamine now even if quartz or granite is chosen, but I will have Teddy look into that issue) and some of the solutions to some of the other items like the DP issue will be taken care of by other existing outstanding issues. Also Sarah Katie Jim and I are hopefully meeting later this week to discuss and provide further direction for you.Thanks, Ted—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by TedClassy1209** [comment]
@sarahhightower It's all good, no need to be sorry. All the issues make sense, it's just that some of them aren't for Nailson to address which is why I wanted us all to discuss before we posted on here so that Nailson has clear direction from one voice on this. We will figure it all out when we meet Thursday though and we will have a good gameplan moving forward. I really appreciate your input.

**Comment by sarahhightower** [comment]
Gotcha gotcha. 

Here's the link to the file I provided the 2D views for:

https://drive.google.com/file/d/1YUrQ8xK1robtglNQW6FK2D5sZsEqFyFz/view?usp=sharing

**Comment by nailsonlandim** [comment]
I've done the adjustments on hash marks. Just one wall is marked, the first one...
![image](https://user-images.githubusercontent.com/70038700/98125427-3bf6fc00-1e93-11eb-88dc-4462764831fd.png)

...and any wall that has a different height. 
![image](https://user-images.githubusercontent.com/70038700/98125885-b6278080-1e93-11eb-9312-f1209e3fd7a7.png)

on standard and custom rooms 
![image](https://user-images.githubusercontent.com/70038700/98128294-7f9f3500-1e96-11eb-8441-436796626011.png)

**Comment by TedClassy1209** [comment]
> Hi @TedClassy1209!
> 
> I've done the adjustments on hash marks. Just one wall is marked, the first one...
> 
> 
> ...and any wall that has a different height.
> 
> 
> on standard and custom rooms
> 

Thanks Nailson, like we talked about yesterday, I think it will be good to (1) just have the first wall's height be notated if the whole room/closet is the same height and then (2) if there are _any_ differing heights within the same closet we would show every one of the closet's wall heights so it is clearer. So, in the second example there should be 4 notations for heights (I'm not sure exactly what changes you made in the room--sloping ceiling, putting a soffit in, etc., but we can work in a separate issue on how to get that information on the plan view).

Like we also talked about, because the height notation of the first wall of each of your examples there is on the top left corner, the hash mark there should go up and to the left rather than up and to the right like it currently is. Then, the 85" wall height notation for Wall C in your second room example should go down and to the right because it is in a bottom right corner. See the following purple hash mark examples:

![wall height hash marks](https://user-images.githubusercontent.com/61025894/98130086-1b20ad80-1e77-11eb-8f80-5a2773cb28fd.png)

Thanks, 

Ted

**Comment by nailsonlandim** [comment]
> Thanks Nailson, like we talked about yesterday, I think it will be good to (1) just have the first wall's height be notated if the whole room/closet is the same height and then (2) if there are _any_ differing heights within the same closet we would show every one of the closet's wall heights so it is clearer. So, in the second example there should be 4 notations for heights (I'm not sure exactly what changes you made in the room--sloping ceiling, putting a soffit in, etc., but we can work in a separate issue on how to get that information on the plan view).

Thanks, in fact that makes things easier. 

> Like we also talked about, because the height notation of the first wall of each of your examples there is on the top left corner, the hash mark there should go up and to the left rather than up and to the right like it currently is. Then, the 85" wall height notation for Wall C in your second room example should go down and to the right because it is in a bottom right corner. See the following purple hash mark examples:

Got it,

**Comment by TedClassy1209** [comment]
Here is how the hash marks for each corner of your 3rd example should look if you were to identify the height of each wall (and depending how the walls were collected):

![wall height hash marks 2](https://user-images.githubusercontent.com/61025894/98131388-74d5a780-1e78-11eb-9a70-bb9712661655.png)

**Comment by nailsonlandim** [comment]
Hi @TedClassy1209 , 

I could adjust most of hash marks as asked. 

Standard rooms are fine:

![image](https://user-images.githubusercontent.com/70038700/98175105-bd22b300-1ed4-11eb-9bb1-a7a89297f27b.png)

![image](https://user-images.githubusercontent.com/70038700/98175232-f5c28c80-1ed4-11eb-92d6-b2e1360abaca.png)

About Custom Rooms

I am working on a solution for the inner ones that aren't as you specified. 
I compare the angle of the current wall to the next one and based on that I set a different angle on the hashmark.
 
![image](https://user-images.githubusercontent.com/70038700/98173651-2f45c880-1ed2-11eb-9b98-2d632b4d2f8c.png)

Tell me what you think.

**Comment by TedClassy1209** [comment]
Hi Nailson, thanks for your work on this. Once you get those inner corner wall heights having the hash pointing the correct dimension it looks good. Can you also have the font size match the size of the wall lengths and the section depths/widths? It looks bold and smaller. Thanks, Ted

**Comment by TedClassy1209** [comment]
Can you explain why the top room of your examples has a 119" top wall (Wall B) and a 120" bottom wall (Wall D)? My guess is both are actually the default 120". We have noticed that the dimensions are not always accurate now and we aren't sure why..

**Comment by nailsonlandim** [comment]
> Hi Nailson, thanks for your work on this. Once you get those inner corner wall heights having the hash pointing the correct dimension it looks good. Can you also have the font size match the size of the wall lengths and the section depths/widths? EDIT: It looks bolder in each room and then smaller or larger than the other fonts depending on the size of the room. Thanks, Ted

I can tweak it to be as similar as possible. I used Text objects instead because I can hide them

**Comment by nailsonlandim** [comment]
> Can you explain why the top room of your examples has a 119" top wall (Wall B) and a 120" bottom wall (Wall D)? My guess is both are actually the default 120". We have noticed that the dimensions are not always accurate now and we aren't sure why..

I will take a look at that, Thanks!

**Comment by nailsonlandim** [comment]
Hi @TedClassy1209 , I've finished the hash marks for custom rooms, take a look. 

I created an odd synthetic project to force it 

**Large custom room**
![image](https://user-images.githubusercontent.com/70038700/98389447-d6da0c80-2032-11eb-9383-b23e8d25df04.png)

**Synthetic test custom room**
![image](https://user-images.githubusercontent.com/70038700/98390052-a8106600-2033-11eb-87c3-b14fbae1e8a3.png)

Check it at: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

**Comment by nailsonlandim** [comment]
Hi @TedClassy1209 @sarahhightower @kategodfrey @JBTattersall 

I've performed the following e-mail list fixes:

**Item 2** 3 elevation export still has too much white space
![image](https://user-images.githubusercontent.com/70038700/98701546-292f6c00-2358-11eb-865a-eb55abfe76e5.png)
Fixed also the 2 elevations per page

**Item 5** Those door hole numbers need to be centered on that right door and it only needs to be on the right door of a double door section, not on both doors.
![image](https://user-images.githubusercontent.com/70038700/98703243-0f8f2400-235a-11eb-8bae-3cedb456afbe.png)

**Item 8** Left exposed TS should have the hash point up and to the left (this needs to be changed), right exposed should be up and to the right (like it currently is for all)
![image](https://user-images.githubusercontent.com/70038700/98703425-3b120e80-235a-11eb-87ae-dadff19d87e7.png)

**Item 15** Change hamper label—it should simply say Hamper (not wire or canvas or whatever…we just want it to say that it’s a hamper—center it vertically and horizontally on the hamper face

![image](https://user-images.githubusercontent.com/70038700/98703838-ba9fdd80-235a-11eb-8235-d835c5114b13.png)

The TK is an ongoing work, along with **Item 4** _(So many different fonts for different dimensions/annotations (some bold, some bubble text, some good normal text)--they need to be uniform and the good normal text. The hash marks need to be the same as well--a thinner line.)_ 

As is a Text I can place it with more precision than a regular OpenGL Dimension. Let me know if it looks well.

**Item 9** The TK height needs to be centered vertically on the TK (right now it is skewed above center).
![image](https://user-images.githubusercontent.com/70038700/98703505-52e99280-235a-11eb-86d2-6d2de651c04b.png)

**Comment by nailsonlandim** [comment]
Hi @TedClassy1209 @sarahhightower @kategodfrey @JBTattersall 

Fixed

**Item 7** - Countertop labels need to be centered above the countertop

_Render of 2D Views project_
![image](https://user-images.githubusercontent.com/70038700/98746097-8ac0fc00-2393-11eb-9d79-2c94b5c279d8.png)

_Render of Large Custom Room_
![image](https://user-images.githubusercontent.com/70038700/98746705-c0b2b000-2394-11eb-9f51-359eb653c394.png)



I've applied the same requirements for Exposed Labels asked on Item 8 (Topshelf)

I also fixed the Countertop Depth of **Item 1** - _(incorrect CT depth (CT is 21.50 deep—20” section depth plus 1.5” overhang)_

**Comment by nailsonlandim** [comment]
Hi @TedClassy1209 @sarahhightower @kategodfrey @JBTattersall 

**Fixed** - **Item 12** (_Different widths for uppers and lowers (See Wall B)—even when both using normal partitions (not using uppers)—right now it just puts the dimensions on eachother below the elevation—can we have it recognize this situations and put the higher section’s width dimension right below or right above it rather than on top of where the lower section’s width dimension is?_)

It finds if there is more than one hanging opening at the wall and has different heights, so it changes the dimensions. the upper dimension above the lower. 
I can set them otherwise (lower above upper)

_Render of 2D Views project_
![image](https://user-images.githubusercontent.com/70038700/98859688-4e4bd980-2441-11eb-97e0-13ffc064588e.png)

_Render of Large Custom Room_
![image](https://user-images.githubusercontent.com/70038700/98859301-ba7a0d80-2440-11eb-9b55-1a23a0ff9cdd.png)

Available at: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

**Comment by TedClassy1209** [comment]
> Hi @TedClassy1209 @sarahhightower @kategodfrey @JBTattersall
> 
> **Fixed** - **Item 12** (_Different widths for uppers and lowers (See Wall B)—even when both using normal partitions (not using uppers)—right now it just puts the dimensions on eachother below the elevation—can we have it recognize this situations and put the higher section’s width dimension right below or right above it rather than on top of where the lower section’s width dimension is?_)
> 
> It finds if there is more than one hanging opening at the wall and has different heights, so it changes the dimensions. the upper dimension above the lower.
> I can set them otherwise (lower above upper)
> 
> _Render of 2D Views project_
> ![image](https://user-images.githubusercontent.com/70038700/98859688-4e4bd980-2441-11eb-97e0-13ffc064588e.png)
> 
> _Render of Large Custom Room_
> ![image](https://user-images.githubusercontent.com/70038700/98859301-ba7a0d80-2440-11eb-9b55-1a23a0ff9cdd.png)
> 
> Available at: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

Thanks Nailson. A few things on this one:

1. Is there any way of putting the upper width right above the upper opening rather than down below when there are different widths between the upper and the lower? That would be preferred. See the following:

![image](https://user-images.githubusercontent.com/61025894/98862630-76790f80-2424-11eb-807d-9c5c14315732.png)

2. If the widths of the upper and lower are the same, it should just show the one dimension.
3. In your first example (Render of 2D Views project), it has 33.75 and 35.25 shown on the right side of the elevation--it should only be 33.75" as there is only one opening there. The 35.25 might be the distance between the lower partition and the wall, but we don't need that dimension, I'm not sure where it is coming from. That 85.5" on the lower left of that same elevation also needs to be removed--I think that was a different issue in that same email I sent.

Let me know if there are any questions

**Comment by TedClassy1209** [comment]
> Hi @TedClassy1209 @sarahhightower @kategodfrey @JBTattersall
> 
> I've performed the following e-mail list fixes:
> 
> **Item 2** 3 elevation export still has too much white space
> ![image](https://user-images.githubusercontent.com/70038700/98701546-292f6c00-2358-11eb-865a-eb55abfe76e5.png)
> Fixed also the 2 elevations per page
> 
> **Item 5** Those door hole numbers need to be centered on that right door and it only needs to be on the right door of a double door section, not on both doors.
> ![image](https://user-images.githubusercontent.com/70038700/98703243-0f8f2400-235a-11eb-8bae-3cedb456afbe.png)
> 
> **Item 8** Left exposed TS should have the hash point up and to the left (this needs to be changed), right exposed should be up and to the right (like it currently is for all)
> ![image](https://user-images.githubusercontent.com/70038700/98703425-3b120e80-235a-11eb-87ae-dadff19d87e7.png)
> 
> **Item 15** Change hamper label—it should simply say Hamper (not wire or canvas or whatever…we just want it to say that it’s a hamper—center it vertically and horizontally on the hamper face
> 
> ![image](https://user-images.githubusercontent.com/70038700/98703838-ba9fdd80-235a-11eb-8235-d835c5114b13.png)
> 
> The TK is an ongoing work, along with **Item 4** _(So many different fonts for different dimensions/annotations (some bold, some bubble text, some good normal text)--they need to be uniform and the good normal text. The hash marks need to be the same as well--a thinner line.)_
> 
> As is a Text I can place it with more precision than a regular OpenGL Dimension. Let me know if it looks well.
> 
> **Item 9** The TK height needs to be centered vertically on the TK (right now it is skewed above center).
> ![image](https://user-images.githubusercontent.com/70038700/98703505-52e99280-235a-11eb-86d2-6d2de651c04b.png)
> 
> EDIT: Everything is available at the same branch, https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

This looks good Nailson, thank you. A few things on these ones as well:

1. The TK height dimension should still be normal-horizontally written, it doesn't need to be parallel with vertical line.
2. For the Hamper label--I am wondering why the KD label, Hamper label, and the handle don't all line up with each other--they should all be centered on the face horizontally, so they should all line up with each other. I can't tell which one(s) are off? Looking at the other elevations, it looks like maybe the KD label isn't centered? However, the Hamper label isn't lined up with the handle either, so perhaps there are more that aren't centered?
3. This is kind of nitpicky, but the CT hash mark looks good in that it goes to the top of the CT. However, the topshelf hash mark goes to the bottom of the topshelf--can you have the topshelf hashmark(s) go to the top of the topshelf rather than the bottom of it?

Thanks, 

Ted

**Comment by nailsonlandim** [comment]
Hi @TedClassy1209 , thanks for the feedback!

> 1. Is there any way of putting the upper width right above the upper opening rather than down below when there are different widths between the upper and the lower? That would be preferred. See the following:

I`ll try, It would be better.

> 2. If the widths of the upper and lower are the same, it should just show the one dimension.

Got it

> In your first example (Render of 2D Views project), it has 33.75 and 35.25 shown on the right side of the elevation--it should only be 33.75" as there is only one opening there. The 35.25 might be the distance between the lower partition and the wall, but we don't need that dimension, I'm not sure where it is coming from.

Understood.

> That 85.5" on the lower left of that same elevation also needs to be removed--I think that was a different issue in that same email I sent.

Yes, It`s Item 1

**Comment by nailsonlandim** [comment]
Thanks for the feedback @TedClassy1209 
> 1. The TK height dimension should still be normal-horizontally written, it doesn't need to be parallel with vertical line.

Got it 

> 2. For the Hamper label--I am wondering why the KD label, Hamper label, and the handle don't all line up with each other--they should all be centered on the face horizontally, so they should all line up with each other. I can't tell which one(s) are off? Looking at the other elevations, it looks like maybe the KD label isn't centered? However, the Hamper label isn't lined up with the handle either, so perhaps there are more that aren't centered?

I`ll look at that too

> 3. This is kind of nitpicky, but the CT hash mark looks good in that it goes to the top of the CT. However, the topshelf hash mark goes to the bottom of the topshelf--can you have the topshelf hashmark(s) go to the top of the topshelf rather than the bottom of it?

In fact, I don't think it's nitpicky, it's important to maintain consistency on a project. It`s an easy adjustment. Thanks!

**Comment by nailsonlandim** [comment]
Hi @TedClassy1209 @sarahhightower @kategodfrey @JBTattersall 

### Adressed Items:
> The TK height dimension should still be normal-horizontally written, it doesn't need to be parallel with vertical line.

**Done:**
![image](https://user-images.githubusercontent.com/70038700/99452233-0ece3300-2902-11eb-9bab-c25e61f148b0.png)

> 1. Is there any way of putting the upper width right above the upper opening rather than down below when there are different widths between the upper and the lower? That would be preferred. See the following:

**Done, also applied it to plan views,** 

![image](https://user-images.githubusercontent.com/70038700/99451247-b2b6df00-2900-11eb-9b13-ee5825ca250b.png)

> 1. If the widths of the upper and lower are the same, it should just show the one dimension.

**Done.**

![image](https://user-images.githubusercontent.com/70038700/99536752-84311680-2989-11eb-8d47-3dde4c4e95b5.png)

![image](https://user-images.githubusercontent.com/70038700/99536250-de7da780-2988-11eb-8ef6-a19d754fe32e.png)


> 3. This is kind of nitpicky, but the CT hash mark looks good in that it goes to the top of the CT. However, the topshelf hash mark goes to the bottom of the topshelf--can you have the topshelf hashmark(s) go to the top of the topshelf rather than the bottom of it?

**Topshelf**
![image](https://user-images.githubusercontent.com/70038700/99452604-82704000-2902-11eb-9e89-d466a3e32272.png)

**Coutertop**
![image](https://user-images.githubusercontent.com/70038700/99452820-d0854380-2902-11eb-92b4-01c3af03061a.png)

### Unadressed Item:
> For the Hamper label--I am wondering why the KD label, Hamper label, and the handle don't all line up with each other--they should all be centered on the face horizontally, so they should all line up with each other. I can't tell which one(s) are off? Looking at the other elevations, it looks like maybe the KD label isn't centered? However, the Hamper label isn't lined up with the handle either, so perhaps there are more that aren't centered?

After you say that I checked it and it really does not align perfectly, but I think we can't do much without lots of work. 

Mathematically they are in the middle but depending on how far from Point of View it gets rendered it displaces from the middle. So even if I tweak it there's no guarantee that it will work on every scenario.

Available at: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

**Comment by nailsonlandim** [comment]
Hi @TedClassy1209 @sarahhightower @kategodfrey @JBTattersall 

**I've added the section depths and hash marks, and if shelves differ from the section depth, it shows every shelf dimension.**

![image](https://user-images.githubusercontent.com/70038700/100102915-0070a180-2e43-11eb-8e87-eb9a6e45cab5.png)

![image](https://user-images.githubusercontent.com/70038700/100103912-44b07180-2e44-11eb-851f-2f8aa5760380.png)

![image](https://user-images.githubusercontent.com/70038700/100103145-44fc3d00-2e43-11eb-8522-156d48a59efa.png)

**I've added also the glass shelves dimensions, with their thickness.**

![image](https://user-images.githubusercontent.com/70038700/100102995-1c744300-2e43-11eb-98b3-af5298c02f52.png)

Right now I am working on add labels to Slanted Shoe Shelves, Flat Crowns, Valences, and Light rails

Available at: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23164-2dview-enhancements

**Comment by TedClassy1209** [comment]
Unsure if this is entirely applicable because of other changes other branches, but I got this following error after Preparing 2D Views in the 164 branch:

![Traceback errors](https://user-images.githubusercontent.com/61025894/100643582-7c616280-32f7-11eb-9a12-56e7b69bbce2.png)

Did not happen in the 226 2D cleanup branch.

**Comment by TedClassy1209** [comment]
Valance is context is spelled valance--not valence--unsure if that is the issue?

**Comment by nailsonlandim** [comment]
Sorry for the typo, got it fixed. 

Especially for this part, I created a different branch so I could save the work elsewhere https://github.com/nailsonlandim/SNaP-Dev/tree/valances_n_lightrails.

I'll merge it back once it's ready

**Comment by nailsonlandim** [comment]
Hi @TedCC1209 

- Finished Valance and Light Rails
- Removed dimensions on Plan View
- Merged it to issue_#226-2dview-cleanup including @ClassyTeddy work at [Flat-Crown-and-Crown-Molding-Improvements](https://github.com/CCDevNelson/SNaP-Dev/tree/Flat-Crown-and-Crown-Molding-Improvements)

![image](https://user-images.githubusercontent.com/70038700/100806958-34822e80-3410-11eb-8dea-9f029b01b793.png)

Available at: https://github.com/nailsonlandim/SNaP-Dev/tree/issue_%23226-2dview-cleanup

Everything is in only one branch. and ready to start receiving the labels on Plan View.
